sharding/contracts/sharding_manager.sol

```diff
@@ -1,253 +1,326 @@
 pragma solidity ^0.4.19;
 
 contract SMC {
-  event TxToShard(address indexed to, int indexed shardId, int receiptId);
-  event CollationAdded(int indexed shardId, uint expectedPeriodNumber, 
-                     bytes32 periodStartPrevHash, bytes32 parentHash,
-                     bytes32 transactionRoot, address coinbase,
-                     bytes32 stateRoot, bytes32 receiptRoot,
-                     int number, bool isNewHead, int score);
-  event Deposit(address collator, int index);
-  event Withdraw(int index);
-
-  struct Collator {
-    // Amount of wei the collator holds
-    uint deposit;
-    // The collator's address
-    address addr;
+  event HeaderAdded(uint indexed shard_id, bytes32 parent_hash,
+                    bytes32 chunk_root, int128 period, int128 height,
+                    address proposer_address, uint proposer_bid,
+                    bytes proposer_signature);
+  event NotaryRegistered(address notary, uint pool_index);
+  event NotaryDeregistered(address notary, uint pool_index, uint deregistered_period);
+  event NotaryReleased(address notary, uint pool_index);
+  event ProposerRegistered(uint pool_index);
+  event ProposerDeregistered(uint index);
+  event ProposerReleased(uint index);
+
+  // Entry in notary_registry
+  struct Notary {
+    // When notary asked for unregistration
```

Comment:
When? So this is a block number? Period number? We should rename this property to be more representative of what it does because it sounds like a boolean but your comment says "when".